sunpy/map/mapbase.py

```diff
@@ -1891,6 +1893,77 @@ def draw_limb(self, axes=None, **kwargs):
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate.
+        Parameters
+        ----------
+        bottom_left : `~astropy.coordinates.SkyCoord`
+            The bottom-left coordinate of the rectangle. It can
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the rectangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the rectangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the rectangle. Required if ``top_right`` is omitted.
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the rectangle, defaults to the current
+            axes.
+        Returns
+        -------
+        rect : `list`
+            A list containing the `~astropy.visualization.wcsaxes.patches.Quadrangle` object, after
+            it has been added to ``axes``.
+        Notes
+        -----
+        If user's astropy version is < 4.2, it defaults to using draw_rectangle instead
+        Extra keyword arguments to this function are passed through to the
+        `~astropy.visualization.wcsaxes.patches.Quadrangle` instance.
+        """
+        if version.parse(astropy.__version__) < version.parse("4.2.0"):
+            warnings.warn("draw_quardrangle is a feature of astropy > 4.2, using draw_rectangle")
+            self.draw_rectangle(bottom_left=bottom_left, width=width, height=height,
+                                axes=axes, top_right=top_right, **kwargs)
+        else:
+            if isinstance(top_right, u.Quantity) and isinstance(width, u.Quantity):
+                # The decorator assigns the first positional arg to top_right and so on.
+                height = width
+                width = top_right
+                top_right = None
+
+            bottom_left, top_right = get_rectangle_coordinates(
+                bottom_left, top_right=top_right, width=width, height=height)
+
+            width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
+            height = Latitude(top_right.spherical.lat - bottom_left.spherical.lat)
+
+            if not axes:
+                axes = plt.gca()
+            if wcsaxes_compat.is_wcsaxes(axes):
+                axes_unit = u.deg
+            else:
+                axes_unit = self.spatial_units[0]
+
+            coord = bottom_left.transform_to(self.coordinate_frame)
+            kwergs = {
+                "transform": wcsaxes_compat.get_world_transform(axes),
+                "color": "white",
+                "fill": False,
+            }
+            kwergs.update(kwargs)
+            rect = Quadrangle(
+                (bottom_left.Tx, bottom_left.Ty) * u.arcsec, width, height, **kwergs)
+            axes.add_artist(rect)
+            return [rect]
```

Comment:
What's the motivation for returning a list instead of just `rect`?

Reply:
This was based on the fact that draw_rectangle returns a list as well, I'm a little unclear as to why we're returning `rect` when it's being plotted on the axes though.